include/aspect/particle/property/cpo_bingham_average.h

```diff
@@ -0,0 +1,202 @@
+/*
+ Copyright (C) 2022 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file LICENSE.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_particle_property_cpo_s_wave_anisotropy_h
+#define _aspect_particle_property_cpo_s_wave_anisotropy_h
+
+#include <aspect/particle/property/interface.h>
+#include <aspect/simulator_access.h>
+#include <array>
+#include <random>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+
+      /**
+       * Computes the Bingham average of the CPO particle properties.
+       * See https://courses.eas.ualberta.ca/eas421/lecturepages/orientation.html for more info.
+       *
+       * The layout of the data vector per partcle is the following (note that for this plugin the following dim's are always 3):
+       * 1 averaged a axis of olivine -> 3 (dim) doubles, starts at:
+       *                                   data_position + 1,
+       * 2 averaged b axis of olivine -> 3 (dim) doubles, starts at:
+       *                                   data_position + 4
+       * 3 averaged c axis of olivine -> 3 (dim) doubles, starts at:
+       *                                   data_position + 7
+       * 4 averaged a axis of enstatite -> 3 (dim) doubles, starts at:
+       *                                    data_position + 10
+       * 5 averaged b axis of enstatite -> 3 (dim) doubles, starts at:
+       *                                    data_position + 13
+       * 6 averaged c axis of enstatite -> 3 (dim) doubles, starts at:
+       *                                    data_position + 16
+       *
+       * @ingroup ParticleProperties
+       */
+      template <int dim>
+      class CpoBinghamAverage : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * constructor
+           */
+          CpoBinghamAverage();
+
+          /**
+           * Initialization function. This function is called once at the
+           * beginning of the program after parse_parameters is run.
+           */
+          virtual
+          void
+          initialize ();
```

Comment:
```suggestion
          void
          initialize () override;
```
Almost the same, but better practice that we already do in all other plugins (yours were probably forked before we made the change). See here for a discussion why if you are interested: https://github.com/geodynamics/aspect/issues/3258